lib/livebook_web/live/output/frame_component.ex

```diff
@@ -14,75 +13,47 @@ defmodule LivebookWeb.Output.FrameComponent do
 
     socket = assign(socket, assigns)
 
-    socket =
-      if socket.assigns.counter == 0 do
-        assign(socket,
-          counter: 1,
-          output_count: length(outputs),
-          persistent_id_map: map_idx_to_persistent_id(outputs, socket.assigns.id)
-        )
-      else
-        socket
-      end
+    socket = assign_new(socket, :num_outputs, fn -> length(outputs) end)
 
     socket =
       case update_type do
         nil ->
-          assign(socket, outputs: outputs)
+          stream(socket, :outputs, stream_items(outputs))
 
         :replace ->
-          prev_output_count = socket.assigns.output_count
-          prev_persistent_id_map = socket.assigns.persistent_id_map
-
-          output_count = length(outputs)
-          persistent_id_map = map_idx_to_persistent_id(outputs, socket.assigns.id)
```

Comment:
Note to self: this was to reuse DOM ids for frame child elements. We did that to fix flickering when animating an image inside a frame in Firefox. With streams the current fix doesn't really work (presumably because of difference in patching "stream" and "append"), so I removed it. The issue is specific to rapid image changes inside frame and to Firefox, so I think it's marginal enough to let it be. My guess is that the flickering is because the `<img>` doesn't have width/height and perhaps Chrome/Safari reads them from the src dataurl before painting while Firefox doesn't, or something along these lines.